### ISSUE TITLE
Fails to start: Windows 8 Pro x64 RTM

### ISSUE THREAD

**Issue by DorpsGek** [OpenTTD/OpenTTD#5359]
**_ratchet2324_ opened the ticket and wrote:**

> Hi all,
> 
> Not sure if I have set the settings correct for the report but:
> \* I have just installed OpenTTD latest stable on Windows 8 Pro x64 RTM and it won't launch, there is just the spinning wheel when I click on it and nothing happens otherwise, when I check in task manager, it appears then disappears almost instantly as if the system won't run it, even though under windows 7 on the same computer it worked perfectly?!?
> 
> Please help!
> 
> Thanks,
> Del

**Reported version**: _1.2.3_
**Operating system**: _Windows_

---

**Comment by DorpsGek** [comment]
**_Transportman_ wrote:**

> There is an issue with 64 bit full screen, see http://www.tt-forums.net/viewtopic.php?f=31&t=63089 . You can disable full screen or run the 32 bit version for now.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11675_

**Comment by DorpsGek** [comment]
**_planetmaker_ wrote:**

> See also http://www.tt-forums.net/viewtopic.php?f=31&t=63089&start=40

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11702_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> I found the problem. It is with ChangeDisplaySettings windows API.
> 
> See http://msdn.microsoft.com/en-us/library/windows/desktop/dd183411(v=vs.85).aspx
> 
> \*** quote from MS Webpage ***
> Note  Apps that you design to target Windows 8 and later can no longer query or set display modes that are less than 32 bits per pixel (bpp); these operations will fail. These apps have a compatibility manifest that targets Windows 8. Windows 8 still supports 8-bit and 16-bit color modes for desktop apps that were built without a Windows 8 manifest; Windows 8 emulates these modes but still runs in 32-bit color mode.
> \*** end quote ***
> 
> In the dev environment, when I set 'settings.dmBitsPerPel' to 32 instead of 8 (in win32_v.cpp, function bool VideoDriver_Win32::MakeWindow(bool full_screen), line 280), OpenTTD goes into full screen mode in Windows 8.
> 
> From what I see (and reading the comments of Blitter_32bppBase not being prefered), there are 2 options:
> 1) Use Blitter_32bppBase despite it not being prefered due to speed issues
> 2) In Blitter_8bppBase, return 32 to GetScreenDepth() when running on Windows 8 or latter.
> 
> I will let the experienced developers make that choice.
> 
> 
> 
> 

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11703_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> Quick extra note, I've modified win32_v.cpp in my checked out copy to add a Windows 8 check to force 32BPP even if using Blitter_8bppBase. It seems to work fine. The check is an extra function (called bool WinSupports8BPP()) added to the top of the file and called from within bool VideoDriver_Win32::MakeWindow(bool full_screen).
> 
> Can I just check that in?

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11704_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> Attached is diff file of my changes. This is for using the 8bit Blitter but using 32bpp.

**Attachments**
- [win32_v.cpp.diff](https://bugs.openttd.org/task/5359/getfile/8712/win32_v.cpp.diff) (1.45 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11705_

**Comment by DorpsGek** [comment]
**_michi_cc_ wrote:**

> The new function should probably use GetVersionEx instead of GetVersion, which is declared deprecated for like 20 years or so :) (Well, VerifyVersionInfo is the "official" function to test OS version, but its usage is more complicated.)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11706_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> @michi_cc No problem. New diff file using VerifyVersionInfo. 

**Attachments**
- [win32_v.cpp_v2.diff](https://bugs.openttd.org/task/5359/getfile/8713/win32_v.cpp_v2.diff) (1.66 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11707_

**Comment by DorpsGek** [comment]
**_michi_cc_ wrote:**

> Hmm, the MSDN page for ChangeDisplaySettings doesn't seem to distinguish i386 and x64 for Windows 8, so I'd assume that it only works on 32-bit Windows by accident (the mentioned compatibility manifest). Going by that I'd limit fullscreen to 32 bpp for all Windows 8 versions, just to make sure it won't get broken later on.
> 
> EDIT: VerifyVersionInfo might be Windows NT line only, MSDN isn't clear as it sometimes simply doesn't mentions older versions, so that would be a problem for the Windows 9x builds (GetVersionEx seems to be safe).

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11708_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> @michi_cc They are both Windows 2000. Prior to that you need to use GetVersion. So which should I use? I may have to add extra checks for those.
> 
> 

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11709_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> This diff removes the x64 check. It will force 32bpp inside of 8bpp Blitter as of Windows 8. Not tested with Win 9x.

**Attachments**
- [win32_v.cpp_v3.diff](https://bugs.openttd.org/task/5359/getfile/8714/win32_v.cpp_v3.diff) (1.57 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11710_

**Comment by DorpsGek** [comment]
**_michi_cc_ wrote:**

> GetVersionEx isn't. Microsoft unfortunately isn't very consistent with accurate version requirements and often drops anything before Windows 2000 when updating documentation.
> 
> The MSDN docs coming with Visual Studio 2008 still list "Requires Windows Vista, Windows XP, Windows 2000 Professional, Windows NT Workstation 3.5 and later, Windows Me, Windows 98, or Windows 95" for GetVersionEx. The line for VerifyVersionInfo is "Requires Windows Vista, Windows XP, or Windows 2000 Professional". I know that even the 2008 docs aren't always right with the requirements, but as VVI is taking the EX variant of the version info, the Windows 2000 is probably right.
> 
> TL;DR: GetVersionEx should be safe on everything we support as long as you use OSVERSIONINFO and not OSVERSIONINFOEX (which is only relevant if you need service pack version).

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11711_

**Comment by DorpsGek** [comment]
**_michi_cc_ wrote:**

> Alternative idea based on the stack trace in the forum thread. Untested as I don't have Windows 8.

**Attachments**
- [display_settings.diff](https://bugs.openttd.org/task/5359/getfile/8715/display_settings.diff) (0.69 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11713_

**Comment by DorpsGek** [comment]
**_gmyx_ wrote:**

> That worked perfectly in my dev environment and much cleaner and simpler too. Both in Win32 and x64.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359#comment11715_

**Comment by DorpsGek** [comment]
**_michi_cc_ closed the ticket.**

**Reason for closing: _Fixed_**

> In r24762. Thanks for the help.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/5359_
